fix(eslint-plugin): [no-output-on-prefix] not reporting failures on alias

rafaelss95 opened the pull request:
With this PR, `alias` like `onChange` in `@Output('onChange') test = new EventEmitter<void>();` are correctly reported.

JamesHenry commented on packages/eslint-plugin/src/rules/no-output-on-prefix.ts:
```diff
@@ -1,47 +1,38 @@
     type: 'suggestion',
     docs: {
-      description: `Name events without the prefix on. See more at https://angular.io/guide/styleguide#dont-prefix-output-properties.`,
+      description: `Ensures that Output names or its alias are not be named "on" or prefixed with it. See more at ${STYLE_GUIDE_LINK}`,
```
```suggestion
      description: `Ensures that Output bindings, including aliases, are not named "on", nor prefixed with it. See more at ${STYLE_GUIDE_LINK}`,
```

JamesHenry commented on packages/eslint-plugin/src/rules/no-output-on-prefix.ts:
```diff
@@ -1,47 +1,38 @@
     messages: {
-      noOutputOnPrefix:
-        'In the class "{{className}}", the output property "{{memberName}}" should not be prefixed with on',
+      noOutputOnPrefix: `Output names or its alias should not be named "on" or prefixed with it (${STYLE_GUIDE_LINK})`,
```
```suggestion
      noOutputOnPrefix: `Output bindings, including aliases, should not be named "on", nor prefixed with it (${STYLE_GUIDE_LINK})`,
```

JamesHenry reviewed (changes requested):
You will also need to run `yarn update-integration-tests` and commit the updated snapshot files within `packages/integration-tests` for this one

rafaelss95 commented:
> You will also need to run `yarn update-integration-tests` and commit the updated snapshot files within `packages/integration-tests` for this one (just the snapshot files though, the rest of the large diff should be ignored)

Yes, I've been trying to do that since yesterday (that's why I converted it to a draft 😆 ). I honestly don't know what happens but I can never execute this command locally without errors 😞 

I literally did everthing: I removed folder did a pull again, ran `yarn i` and `yarn update-integration-tests` and still get this:

```
@angular-eslint/integration-tests: npm ERR! code ETARGET
@angular-eslint/integration-tests: npm ERR! notarget No matching version found for @angular-eslint/builder@9999.0.1-local-integration-tests.
@angular-eslint/integration-tests: npm ERR! notarget In most cases you or one of your dependencies are requesting
@angular-eslint/integration-tests: npm ERR! notarget a package version that doesn't exist.
@angular-eslint/integration-tests: npm ERR! A complete log of this run can be found in:
@angular-eslint/integration-tests: npm ERR!     /Users/<folder>/.npm/_logs/2021-05-18T21_49_04_020Z-debug.log
@angular-eslint/integration-tests: ✖ Package install failed, see above.
@angular-eslint/integration-tests: The Schematic workflow failed. See above.
@angular-eslint/integration-tests: Error: Command failed with exit code 1: npx ng add @angular-eslint/schematics@9999.0.1-local-integration-tests
@angular-eslint/integration-tests:     at makeError (/Users/<folder>/angular-eslint/node_modules/execa/lib/error.js:56:11)
@angular-eslint/integration-tests:     at handlePromise (/Users/<folder>/angular-eslint/node_modules/execa/index.js:114:26)
@angular-eslint/integration-tests:     at processTicksAndRejections (node:internal/process/task_queues:94:5)
@angular-eslint/integration-tests:     at runNgAdd (/Users/<folder>/angular-eslint/packages/integration-tests/integration-tests-setup.ts:119:10)
@angular-eslint/integration-tests:     at setupFixtures (/Users/<folder>/angular-eslint/packages/integration-tests/integration-tests-setup.ts:190:5) {
@angular-eslint/integration-tests:   command: 'npx ng add @angular-eslint/schematics@9999.0.1-local-integration-tests',
@angular-eslint/integration-tests:   exitCode: 1,
@angular-eslint/integration-tests:   signal: undefined,
@angular-eslint/integration-tests:   signalDescription: undefined,
@angular-eslint/integration-tests:   stdout: 'Skipping installation: Package already installed\n' +
@angular-eslint/integration-tests:     '    \n' +
@angular-eslint/integration-tests:     '    All @angular-eslint dependencies have been successfully installed 🎉\n' +
@angular-eslint/integration-tests:     '    \n' +
@angular-eslint/integration-tests:     '    Please see https://github.com/angular-eslint/angular-eslint for how to add ESLint configuration to your project.\n' +
@angular-eslint/integration-tests:     '    \n' +
@angular-eslint/integration-tests:     'UPDATE package.json (1900 bytes)',
@angular-eslint/integration-tests:   stderr: '- Installing packages (npm)...\n' +
@angular-eslint/integration-tests:     'npm ERR! code ETARGET\n' +
@angular-eslint/integration-tests:     'npm ERR! notarget No matching version found for @angular-eslint/builder@9999.0.1-local-integration-tests.\n' +
@angular-eslint/integration-tests:     'npm ERR! notarget In most cases you or one of your dependencies are requesting\n' +
@angular-eslint/integration-tests:     "npm ERR! notarget a package version that doesn't exist.\n" +
@angular-eslint/integration-tests:     '\n' +
@angular-eslint/integration-tests:     'npm ERR! A complete log of this run can be found in:\n' +
@angular-eslint/integration-tests:     'npm ERR!     /Users/<folder>/.npm/_logs/2021-05-18T21_49_04_020Z-debug.log\n' +
@angular-eslint/integration-tests:     '✖ Package install failed, see above.\n' +
@angular-eslint/integration-tests:     'The Schematic workflow failed. See above.',
@angular-eslint/integration-tests:   failed: true,
@angular-eslint/integration-tests:   timedOut: false,
@angular-eslint/integration-tests:   isCanceled: false,
@angular-eslint/integration-tests:   killed: false
@angular-eslint/integration-tests: }
@angular-eslint/integration-tests: error Command failed with exit code 1.
@angular-eslint/integration-tests: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run update-integration-tests exited 1 in '@angular-eslint/integration-tests'
error Command failed with exit code 1.
```

JamesHenry commented:
@rafaelss95 sorry about that, it clearly runs fine on my machine (macOS) and on CI (linux), and I have no way to debug any further for you

JamesHenry commented:
It will not let me push to your remote, here is the update snapshot file you need to apply:

```
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v1123-multi-project-manual-config it should produce the expected lint output 1`] = `
"
Linting \\"v1123-multi-project-manual-config\\"...

__ROOT__/v1123-multi-project-manual-config/src/app/app.component.html
  1:31  error  Invalid binding syntax. Use [(expr)] instead  @angular-eslint/template/banana-in-box

__ROOT__/v1123-multi-project-manual-config/src/app/app.component.spec.ts
  19:6  error  Strings must use singlequote  @typescript-eslint/quotes

__ROOT__/v1123-multi-project-manual-config/src/app/app.component.ts
   2:1  error  'rxjs/Rx' import is restricted from being used. Please import directly from 'rxjs' instead                             no-restricted-imports
   5:1  error  Unexpected console statement                                                                                           no-console
  15:3  error  Lifecycle interface 'OnInit' should be implemented for method 'ngOnInit'. (https://angular.io/styleguide#style-09-01)  @angular-eslint/use-lifecycle-interface

__ROOT__/v1123-multi-project-manual-config/src/app/example.component.ts
   4:13  error  Strings must use singlequote                                                                                                          @typescript-eslint/quotes
  12:3   error  Use @Input rather than the \`inputs\` metadata property (https://angular.io/styleguide#style-05-12)                                     @angular-eslint/no-inputs-metadata-property
  13:3   error  Use @Output rather than the \`outputs\` metadata property (https://angular.io/styleguide#style-05-12)                                   @angular-eslint/no-outputs-metadata-property
  14:3   error  Use @HostBinding or @HostListener rather than the \`host\` metadata property (https://angular.io/styleguide#style-06-03)                @angular-eslint/no-host-metadata-property
  18:3   error  Output bindings, including aliases, should not be named \\"on\\", nor prefixed with it (https://angular.io/guide/styleguide#style-05-16)  @angular-eslint/no-output-on-prefix
   6:35  error  Invalid binding syntax. Use [(expr)] instead                                                                                          @angular-eslint/template/banana-in-box
   8:15  error  Invalid binding syntax. Use [(expr)] instead                                                                                          @angular-eslint/template/banana-in-box
   8:29  error  Invalid binding syntax. Use [(expr)] instead                                                                                          @angular-eslint/template/banana-in-box
   9:48  error  Invalid binding syntax. Use [(expr)] instead                                                                                          @angular-eslint/template/banana-in-box

__ROOT__/v1123-multi-project-manual-config/src/app/example.pipe.ts
  6:14  error  Pipes should implement \`PipeTransform\` interface  @angular-eslint/use-pipe-transform-interface

__ROOT__/v1123-multi-project-manual-config/src/app/multiple-inline-templates.page.ts
   4:13  error  Strings must use singlequote                  @typescript-eslint/quotes
  12:13  error  Strings must use singlequote                  @typescript-eslint/quotes
   6:35  error  Invalid binding syntax. Use [(expr)] instead  @angular-eslint/template/banana-in-box
  14:48  error  Invalid binding syntax. Use [(expr)] instead  @angular-eslint/template/banana-in-box

✖ 19 problems (19 errors, 0 warnings)
  12 errors and 0 warnings potentially fixable with the \`--fix\` option.


Linting \\"another-app\\"...

__ROOT__/v1123-multi-project-manual-config/projects/another-app/src/app/app.component.html
   12:1  error  This line has a length of 159. Maximum allowed is 140   max-len
  306:1  error  This line has a length of 532. Maximum allowed is 140   max-len
  313:1  error  This line has a length of 540. Maximum allowed is 140   max-len
  327:1  error  This line has a length of 219. Maximum allowed is 140   max-len
  328:1  error  This line has a length of 523. Maximum allowed is 140   max-len
  335:1  error  This line has a length of 151. Maximum allowed is 140   max-len
  336:1  error  This line has a length of 288. Maximum allowed is 140   max-len
  347:1  error  This line has a length of 205. Maximum allowed is 140   max-len
  351:1  error  This line has a length of 184. Maximum allowed is 140   max-len
  354:1  error  This line has a length of 220. Maximum allowed is 140   max-len
  358:1  error  This line has a length of 176. Maximum allowed is 140   max-len
  362:1  error  This line has a length of 435. Maximum allowed is 140   max-len
  366:1  error  This line has a length of 176. Maximum allowed is 140   max-len
  379:1  error  This line has a length of 167. Maximum allowed is 140   max-len
  385:1  error  This line has a length of 167. Maximum allowed is 140   max-len
  391:1  error  This line has a length of 167. Maximum allowed is 140   max-len
  397:1  error  This line has a length of 165. Maximum allowed is 140   max-len
  403:1  error  This line has a length of 165. Maximum allowed is 140   max-len
  409:1  error  This line has a length of 165. Maximum allowed is 140   max-len
  429:1  error  This line has a length of 188. Maximum allowed is 140   max-len
  430:1  error  This line has a length of 158. Maximum allowed is 140   max-len
  431:1  error  This line has a length of 165. Maximum allowed is 140   max-len
  432:1  error  This line has a length of 165. Maximum allowed is 140   max-len
  433:1  error  This line has a length of 162. Maximum allowed is 140   max-len
  440:1  error  This line has a length of 187. Maximum allowed is 140   max-len
  441:1  error  This line has a length of 159. Maximum allowed is 140   max-len
  442:1  error  This line has a length of 188. Maximum allowed is 140   max-len
  443:1  error  This line has a length of 174. Maximum allowed is 140   max-len
  450:1  error  This line has a length of 172. Maximum allowed is 140   max-len
  457:1  error  This line has a length of 190. Maximum allowed is 140   max-len
  458:1  error  This line has a length of 160. Maximum allowed is 140   max-len
  462:1  error  This line has a length of 217. Maximum allowed is 140   max-len
  466:1  error  This line has a length of 187. Maximum allowed is 140   max-len
  474:1  error  This line has a length of 190. Maximum allowed is 140   max-len
  475:1  error  This line has a length of 160. Maximum allowed is 140   max-len
  476:1  error  This line has a length of 410. Maximum allowed is 140   max-len
  483:1  error  This line has a length of 2884. Maximum allowed is 140  max-len
  509:1  error  This line has a length of 261. Maximum allowed is 140   max-len
  514:1  error  This line has a length of 230. Maximum allowed is 140   max-len
  518:1  error  This line has a length of 148. Maximum allowed is 140   max-len
  519:1  error  This line has a length of 1046. Maximum allowed is 140  max-len

__ROOT__/v1123-multi-project-manual-config/projects/another-app/src/app/app.component.spec.ts
  19:6  error  Strings must use singlequote  @typescript-eslint/quotes

✖ 42 problems (42 errors, 0 warnings)
  1 error and 0 warnings potentially fixable with the \`--fix\` option.


Linting \\"horse-plants\\"...

All files pass linting.


Linting \\"allergies-fusion\\"...

All files pass linting.


Linting \\"breakfast-cat\\"...

All files pass linting.


Linting \\"prints-soda\\"...

All files pass linting.


Linting \\"ice-cream-monster\\"...

All files pass linting.


Linting \\"shoe-prints\\"...

All files pass linting.


Linting \\"websites-system\\"...

All files pass linting.


Linting \\"robot-websites\\"...

All files pass linting.
"
`;
```

rafaelss95 commented:
> It will not let me push to your remote, here is the update snapshot file you need to apply:

Strange, the "Allow edits and access to secrets by maintainers" box is checked.

In any case, it's working after copying your snippet, thanks!
